Utilities/RelMon/python/utils.py

```diff
@@ -90,11 +90,20 @@ def literal2root (literal,rootType):
 #-------------------------------------------------------------------------------
 
 def getNbins(h):
+  """
+  To be used in loops on bin number with range()
+  In case of 1-d histograms the end-of-range value should be GetNbinsX()+2 
+  In case of 2-d or 3-d histograms the product of GetNbins.()+2 gives the total number of bins, and 1 needs to be added
+  Managed using an offset variable 
+  """
+  offset=0
   biny=h.GetNbinsY()
-  if biny>1:biny+=1
+  if biny>1:
+    offset=1
```

Comment:
PLease verify whether this offset is really needed.
In a 2D histogram there are (NBinsX+2)*(NBinsY+2) bins, including underflow and overflow. That is, they numbers from "0" to "(NBinsX+2)*(NBinsY+2)-1", and to loop through all them the second argument of the (python) range function must be "(NBinsX+2)*(NBinsY+2)", without such an offset.

Reply:
@perrotta right, the point is that the range loops using this output run from 0 up to the output of this function now, which must be (NBinsX+2)(NBinsY+2)+1 to account for the fact that the last element of the range() must be the last item of the list +1. Do you agree?

Reply:
If I am not wrong,(see above)  the last item in the list is "(NBinsX+2)(NBinsY+2)-1" and the last item+1 is "(NBinsX+2)(NBinsY+2)"

Reply:
@perrotta no, I think you are correct, range() starts from 0 but the output of this functions counts from 1, so ok